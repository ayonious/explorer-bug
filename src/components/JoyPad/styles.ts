import styled from 'styled-components';

export const Row = styled.div`
  display: flex;
  justify-content: center;
  width: fit-content;
  margin: auto;
`;
