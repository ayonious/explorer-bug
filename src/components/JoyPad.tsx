import * as React from 'react';
import { Row } from '../styledcomponents/styles';
import { CustomButton } from './Utils';

interface Props {
  handleLeft: any;
  handleRight: any;
  handleGo: any;
}

interface State {}

export default class JoyPad extends React.Component<Props, State> {
  render() {
    return (
      <div>
        <Row>
          <CustomButton onClick={this.props.handleGo} title={'Go'} />
        </Row>
        <Row>
          <CustomButton onClick={this.props.handleLeft} title={'Left'} />
          <CustomButton onClick={this.props.handleRight} title={'Right'} />
        </Row>
      </div>
    );
  }
}
