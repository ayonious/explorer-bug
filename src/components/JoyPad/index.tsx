import * as React from 'react';

import { Row } from './styles';
import CustomButton from '../CustomButton';

interface Props {
  handleLeft: any;
  handleRight: any;
  handleGo: any;
}

const JoyPad = (props: Props) => {
  return (
    <div>
      <Row>
        <CustomButton onClick={props.handleGo} title={'Go'} />
      </Row>
      <Row>
        <CustomButton onClick={props.handleLeft} title={'Left'} />
        <CustomButton onClick={props.handleRight} title={'Right'} />
      </Row>
    </div>
  );
};

export default JoyPad;
