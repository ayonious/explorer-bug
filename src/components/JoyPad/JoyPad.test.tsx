/**
 * @jest-environment jsdom
 */

import { render, screen, fireEvent } from '@testing-library/react';
import { expect } from '@jest/globals';
import React from 'react';
import '@testing-library/jest-dom';

import JoyPad from '.';

describe('JoyPad Component', () => {
  it('renders all control buttons', () => {
    const func = jest.fn();
    render(<JoyPad handleGo={func} handleLeft={func} handleRight={func} />);
    
    expect(screen.getByTestId('move-button-Go')).toBeInTheDocument();
    expect(screen.getByTestId('move-button-Left')).toBeInTheDocument();
    expect(screen.getByTestId('move-button-Right')).toBeInTheDocument();
  });

  it('calls appropriate handlers when buttons are clicked', () => {
    const handleGo = jest.fn();
    const handleLeft = jest.fn();
    const handleRight = jest.fn();

    render(
      <JoyPad
        handleGo={handleGo}
        handleLeft={handleLeft}
        handleRight={handleRight}
      />
    );

    fireEvent.click(screen.getByTestId('move-button-Go'));
    expect(handleGo).toHaveBeenCalledTimes(1);

    fireEvent.click(screen.getByTestId('move-button-Left'));
    expect(handleLeft).toHaveBeenCalledTimes(1);

    fireEvent.click(screen.getByTestId('move-button-Right'));
    expect(handleRight).toHaveBeenCalledTimes(1);
  });

  it('maintains correct button states', () => {
    const func = jest.fn();
    render(<JoyPad handleGo={func} handleLeft={func} handleRight={func} />);

    const goButton = screen.getByTestId('move-button-Go');
    const leftButton = screen.getByTestId('move-button-Left');
    const rightButton = screen.getByTestId('move-button-Right');

    // Check button text content
    expect(goButton).toHaveTextContent('Go');
    expect(leftButton).toHaveTextContent('Left');
    expect(rightButton).toHaveTextContent('Right');

    // Check button accessibility
    expect(goButton).toHaveAttribute('type', 'button');
    expect(leftButton).toHaveAttribute('type', 'button');
    expect(rightButton).toHaveAttribute('type', 'button');
  });
});
