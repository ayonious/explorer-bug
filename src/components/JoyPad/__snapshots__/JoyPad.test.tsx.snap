// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JoyPad renders 1`] = `
<div>
  <div>
    <div
      class="sc-beqWaB cVbQTx"
    >
      <div
        class="sc-gueYoa bGbOlf"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
          data-testid="move-button-Go"
          tabindex="0"
          type="button"
        >
          Go
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class="sc-beqWaB cVbQTx"
    >
      <div
        class="sc-gueYoa bGbOlf"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
          data-testid="move-button-Left"
          tabindex="0"
          type="button"
        >
          Left
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="sc-gueYoa bGbOlf"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
          data-testid="move-button-Right"
          tabindex="0"
          type="button"
        >
          Right
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
